test/unit/any_instance_method_test.rb

```diff
@@ -84,31 +84,19 @@ def test_should_restore_original_method
     method.hide_original_method
     method.define_new_method
     method.remove_new_method
-    method.restore_original_method
```

Comment:
Since this test is named `test_should_restore_original_method` and it seems to have been testing the implementation in the parent class, `StubbedMethod`, presumably we could delete the whole test...?

Reply:
Ah I figured this test was still valid, as the assertion is checking that the original result is returned after `remove_new_method`. `restore_original_method` seems to have been a workaround for older versions of Ruby.

Reply:
Maybe this can replace `test_should_call_remove_new_method`? That only tests that `remove_new_method` is called, but there isn't actually a corresponding unit test for `remove_new_method` itself.

Maybe I'll rename `test_should_restore_original_method` to `test_remove_new_method_restores_original_method`?

Reply:
Ah, yes - that makes sense now. 👍